docs/create_allocator_service.md

```diff
@@ -0,0 +1,435 @@
+# Quickstart Create an Allocator Service
+
+This quickstart describes one way to implement a service which will allocate a Game Server on demand using the Agones API.  This allows a game client or matchmaker service to call the allocator service in order to allocate a Game Server and retrieve its IP address and port number. The game client can then connect directly to the Game Server, so that the minimized latency that is in the Agones design is preserved.
```

Comment:
Personally, i think there are chunks of this that are overkill - I don't think we need to explain Kubernetes - but we can totally link to references that may be useful to understand concepts they may not be familiar with.

That being said - love and appreciate the user empathy in including it, but I think users may find this overwhelming.

Reply:
I'll reduce the amount of text, and focus on linking to documentation over explanation.

Do you have any data on the user base?  Having a clearer understanding of the audience would make writing more targeted documentation easier.  I've been imagining the audience to be the same as that for kubernetes or gke documention, and am trying to replicate the level of detail I see in their docs.

Reply:
I think that's valid - it's for developers for sure - so expect technical expertise.

I think we can assume some K8s knowledge, but should link to docs so that people who aren't that familiar have a reference to go look at.